### PR TITLE
#170122153 Signin with database

### PR DESCRIPTION
**Pull request by Ally4**
#### What does this PR do?

- `Help user to sign in with database in the system`

#### Description of Task to be completed?

- `Create Sign in`

#### How should this be manually tested?

- `When the user sign in by following the instructions in the readme`

#### Any background context you want to provide?

- `Creation of sign in router`

- `Creation of signup router`

#### What are the relevant pivotal tracker stories?

- `https://www.pivotaltracker.com/story/show/170122153`

#### Screenshots (if appropriate)
Screenshot (3067)
Screenshot (3069)

### CODE REVIEW

File: server/Version-2/controllers/signin-signup.js [hunk 1]
```diff
@@ -37,6 +37,39 @@ class SigninSignup {
...
+    if (error) {
+      return res.status(400).json({
+        status: 400,
+        message: error.details[0].message.replace(/"/g, ''),
+      });
+    }
+    if (signin1.rows === 'undefined' || signin1.rows.length === 0) {
+      return res.status(400).json({
+        status: 400,
+        message: 'Same credential must be wrong',
```
Comment (YvesIraguha) [inline server/Version-2/controllers/signin-signup.js]: @Ally4 what does this message means to a user?

File: server/Version-2/controllers/signin-signup.js [hunk 2]
```diff
@@ -37,6 +37,39 @@ class SigninSignup {
...
+      });
+    }
+    const compare = bcrypt.compare(req.body.password, signin1.rows[0].password);
+    if (!compare) {
+      return res.status(404).json({
+        status: 404,
+        message: 'One of the information might not be right for you to signin',
+      });
```
Comment (YvesIraguha) [inline server/Version-2/controllers/signin-signup.js]: @Ally4 what does this message asks the user to do! In addition, the status code here is for `Not found` yet what you meant is returning the invalid credential error. Please search for another corresponding status code, which reflects what's happening.

File: server/Version-2/controllers/signin-signup.js (hunk 1 above)
Comment (Ally4) [reply server/Version-2/controllers/signin-signup.js]: Here the email is not found, I have to look for a lightful message and the right status code

File: server/Version-2/controllers/signin-signup.js (hunk 2 above)
Comment (Ally4) [reply server/Version-2/controllers/signin-signup.js]: Here the password is not correct, I have to look for a lightful message and the right status code

File: server/Version-2/controllers/signin-signup.js (hunk 1 above)
Comment (Ally4) [reply server/Version-2/controllers/signin-signup.js]: This is the change

Screenshot (3070)

File: server/Version-2/controllers/signin-signup.js (hunk 2 above)
Comment (Ally4) [reply server/Version-2/controllers/signin-signup.js]: Here is the other change

Screenshot (3071)

**Pull request merged**